Linux/audit.sh

```diff
@@ -79,10 +79,39 @@ harden() {
 }
 
 
+backup_config_dirs() {
+        # Takes 1 argument - Array of strings of paths that need to be backed up.
+
+        # Get paths that exist
+        local -n dir_arr=$1
+        dir_str=""
+
+        for path in ${dir_arr[@]};
+        do
+                [[ -d $path ]] && dir_str="$dir_str $path"
+                [[ -f $path ]] && dir_str="$dir_str $path"
+        done
+
+        [ -z "$dir_str" ] && echo "Could not backup files. No valid paths in array." && return
+
+        # Make backup in /Backups directory in Home
+        mkdir -p ~/Backups
+        version="$(find ~/Backups -name "$HOSTNAME-config-dir-backup-[0-9]*\.tgz" | wc -l)"
```

Comment:
Does this update the version or just find the old one?

Reply:
This finds the amount of backups there are, which also is the next version number because the initial backup would start at 0.